### ISSUE TITLE
SignInWithEmailDialog: TextFields have hintText, which are not seen by screen-readers

### ISSUE THREAD

**Issue by BenAuerDev** [serverpod/serverpod#2382]
**Describe the bug**
The accessibility report in the pre-launch review (Google PlayStore) flagged the `SignInWithEmailDialog`. I think this is due to the fact, that the TextFields show/have `hintText` instead of `labelText` / `label`

```dart 
TextField(
  enabled: _enabled,
  controller: _emailController,
  keyboardType: TextInputType.emailAddress,
  decoration: InputDecoration(
    hintText: 'Email', // hintText instead of label
    helperText: ' ',
    errorText: _emailIssue,
  ),
  onChanged: (_) {
    setState(() {
      _emailIssue = null;
    });
  },
),
```

**Screenshots**
The text in the screenshot is in German unfortunately but the message/recommendation reads: 
"This object probably has no label which can be read by screen-readers."
 
![pre-launch-report](https://github.com/serverpod/serverpod/assets/136239531/277ba8cd-5a85-49c0-8ee1-5a0d165cd726)

**Serverpod versions**
version 2.0.1

**Platform information**
Android - Google PlayStore

**Additional context**
This is not a major issue as it won't block the release.

**Comment by Isakdl** [comment]
Well spotted! With this issue, we should add an a11y test to find/protect against all these accessibility issues.

**Comment by marcelomendoncasoares** [comment]
Closing since this is related to the legacy auth, which is no longer receiving updates.
